solutions/usaco-991.mdx

```diff
@@ -0,0 +1,72 @@
+---
+id: usaco-991
+source: USACO Silver 2020 January
+title: Loan Repayment
+author: Brad Ma
+---
+
+[Official Analysis (C++)](http://www.usaco.org/current/data/sol_loan_silver_jan20.html)
+
+## Implementation
+**Time Complexity:** $\mathcal{O}(\sqrt{N} \cdot \log{N})$
+
+<LanguageSection>
+  
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class LoanRepayment {
+	// our checking function in the binary search
+	static boolean isOk (long numGallons, long withinDays, long atLeast, long x) {
+		long g = 0;
+		while (withinDays > 0 && g < numGallons) {
+			long y = (numGallons - g) / x;
+			if (y < atLeast) {
+				long leftover = ((numGallons - g) + (atLeast - 1)) / atLeast;
+				return leftover <= withinDays;
+			}
+			long maxMatch = numGallons - (x * y);
+			long numDays = (maxMatch - g) / y + 1;
+			if (numDays > withinDays) {  // take a maximum
+				numDays = withinDays;
+			}
+			g += y * numDays;  // update values
+			withinDays -= numDays;
+		}
+		return g >= numGallons;
+	}
+
+	public static void main (String[] args) throws IOException {
+		Kattio io = new Kattio("loan");
+		long numGallons = io.nextLong();
+		long withinDays = io.nextLong();
+		long atLeast = io.nextLong();
+
+		long low = 1;
+		long high = 1000000000000L;
```

Comment:
i'd do integer.max_value but sure

Reply:
you need 10^12, not integer max value for this problem